src/test/groovy/com/synacy/lesson03/exercise/EnrollmentServiceSpec.groovy

```diff
@@ -0,0 +1,106 @@
+package com.synacy.lesson03.exercise
+
+import com.synacy.lesson03.exercise.domain.CourseClass
+import com.synacy.lesson03.exercise.domain.EnrollmentDto
+import com.synacy.lesson03.exercise.domain.Printable
+import com.synacy.lesson03.exercise.domain.Student
+import com.synacy.lesson03.exercise.domain.StudentEmailType
+import com.synacy.lesson03.exercise.domain.StudentProfile
+import com.synacy.lesson03.exercise.domain.StudyLoad
+
+/**
+ * Created by kenichigouang on 5/4/17.
+ */
+class EnrollmentServiceSpec extends spock.lang.Specification {
+
+    EnrollmentService service
+
+    StudentProfileService studentProfileService
+    CourseClassService courseClassService
+    EnrollmentNotificationService enrollmentNotificationService
+    StudyLoadFormatter studyLoadFormatter
+    SystemService systemService
+
+    void setup() {
+        service = new EnrollmentService()
+
+        studentProfileService = Mock()
+        courseClassService = Mock()
+        enrollmentNotificationService = Mock()
+        studyLoadFormatter = Mock()
+        systemService = Mock()
+
+        service.studentProfileService = studentProfileService
+        service.courseClassService = courseClassService
+        service.enrollmentNotificationService = enrollmentNotificationService
+        service.studyLoadFormatter = studyLoadFormatter
+        service.systemService = systemService
+    }
+
+    def "ProcessEnrollment should update student profile on processing on enrollment"() {
+        given:
+        EnrollmentDto enrollmentDetails = Mock()
+        StudentProfile studentProfile = Mock()
+
+        enrollmentDetails.getStudentProfile() >> studentProfile
+        enrollmentDetails.getEnrolledClasses() >> []
+
+        when:
+        service.processEnrollment(enrollmentDetails)
+
+        then:
+        1 * studentProfileService.updateStudentProfile(studentProfile)
+    }
+
+    def "ProcessEnrollment should enroll student to a class/classes on processing of enrollment"() {
+        given:
+        EnrollmentDto enrollmentDetails = Mock()
+        Student student = Mock()
+
+        CourseClass courseClass = Mock()
+
+        def enrolledClasses = [
+                courseClass
+        ]
```

Comment:
Testing with collections is better when you have at least two. For this instance, we would like to gain confidence that **_all_** `CourseClass`es within `EnrollmentDto` are enrolled by the student. In that case, we would not be sure if all classes in the collection were enrolled if we only have 1 in our test case scenario.